app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt

```diff
@@ -206,17 +211,431 @@ class StateFragmentTest {
+      clickContinueInteractionButton()
+
+      // Attempt to submit an wrong answer.
+      typeFractionText("1/3")
+      clickSubmitAnswerButton()
+
+      // Attempt to submit an wrong answer twice.
+      typeFractionText("1/9")
+      clickSubmitAnswerButton()
+
+      scrollToViewType(CONTENT)
+      val htmlResult =
+        "What fraction represents half of something?"
+      onView(atPositionOnView(R.id.state_recycler_view, 0, R.id.content_text_view)).check(
+        matches(
+          withText(htmlResult)
+        )
+      )
+
+      scrollToViewType(FEEDBACK)
+      onView(withId(R.id.feedback_text_view)).check(matches(isDisplayed()))
+      onView(withId(R.id.fraction_input_interaction_view)).check(matches(isDisplayed()))
+    }
+  }
+
+  @Test
+  fun testStateFragment_loadExp_submitWrongAnswerTwice_configChange_feedbackAndQuestionIsVisible() {
+    launchForExploration(TEST_EXPLORATION_ID_2).use {
+      startPlayingExploration()
+      clickContinueInteractionButton()
+
+      // Attempt to submit an wrong answer.
+      typeFractionText("1/3")
+      clickSubmitAnswerButton()
+
+      // Attempt to submit an wrong answer twice.
+      typeFractionText("1/9")
+      clickSubmitAnswerButton()
+
+      rotateToLandscape()
+
+      scrollToViewType(CONTENT)
+      val htmlResult =
+        "What fraction represents half of something?"
+      onView(atPositionOnView(R.id.state_recycler_view, 0, R.id.content_text_view)).check(
+        matches(
+          withText(htmlResult)
+        )
+      )
+
+      scrollToViewType(FEEDBACK)
+      onView(withId(R.id.feedback_text_view)).check(matches(isDisplayed()))
+      onView(withId(R.id.fraction_input_interaction_view)).check(matches(isDisplayed()))
+    }
+  }
+
   //  4. Configuration change with typed text (e.g. for numeric or text input) retains that
   //     temporary
   //     text and you can continue with the exploration after rotating.
+
+  @Test
+  @Ignore("Work on issue regarding this test is in progress")
+  fun testStateFragment_loadExp_submitAnswer_configChange_explorationContinues() {
+    launchForExploration(TEST_EXPLORATION_ID_2).use {
+      startPlayingExploration()
+      clickContinueInteractionButton()
+
+      // Attempt to submit an wrong answer.
+      typeFractionText("1/2")
+
+      rotateToLandscape()
+      clickSubmitAnswerButton()
+
+      clickContinueInteractionButton()
+    }
+  }
+
   //  5. Configuration change after submitting the wrong answer to show that the old answer & re-ask
   //     of the question stay the same.
+  @Test
+  fun testStateFragment_loadExp_submitWrongAnswer_configChange_feedbackAndQuestionVisible() {
+    launchForExploration(TEST_EXPLORATION_ID_2).use {
+      startPlayingExploration()
+      clickContinueInteractionButton()
+
+      // Attempt to submit an wrong answer.
+      typeFractionText("1/3")
+      clickSubmitAnswerButton()
+
+      rotateToLandscape()
+
+      scrollToViewType(CONTENT)
+      val htmlResult =
```

Comment:
let's test this `htmlResult` before rotating too. In this way, we are testing that the question is the same before/after rotation.

Reply:
Done.